tests/usbredir_test.go

```diff
@@ -105,14 +107,34 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 		It("Should fail when limit is reached", func() {
 			var tests [v1.UsbClientPassthroughMaxNumberOf + 1]testCase
 			for i := 0; i <= v1.UsbClientPassthroughMaxNumberOf; i++ {
-				ctx, cancelFn := context.WithCancel(context.Background())
-				tests[i] = testCase{
-					cancel: cancelFn,
-					err:    make(chan error),
+			retry_loop:
+				for try := 0; try < 3; try++ {
+					ctx, cancelFn := context.WithCancel(context.Background())
+					tests[i] = testCase{
+						cancel:  cancelFn,
+						connect: make(chan struct{}),
+						err:     make(chan error),
+					}
+					ctx = context.WithValue(ctx, "connected", tests[i].connect)
+					go runConnectGoroutine(virtClient, name, namespace, ctx, tests[i].err)
+
+					if i == v1.UsbClientPassthroughMaxNumberOf {
+						// Last test is meant to fail.
+						break
+					}
+
+					// Till the last test, all sockets must be connected
+					select {
+					case <-tests[i].connect:
+						break retry_loop
+					case err := <-tests[i].err:
+						log.Log.Reason(err).Infof("Failed early. Try again (%d)", try)
+					case <-time.After(time.Second):
+						log.Log.Infof("Took too long. Try again (%d)", try)
+						tests[i].cancel()
```

Comment:
Where do we ensure we actually succeeded?

Reply:
Taking your comment in the context of this change, so we succeeded if `connect` chan received, that is, this `try` for establishing a TCP connection succeeded. At that point we already sent/receive the `hello` capabilities message which is all this test would read/write.